File: 27 Remove Element/removeElement.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0;
        
        for(int i = 0; i<nums.size(); i++){
            if(nums[i] != val)
                if(i != k){
                    swap(nums[i], nums[k++]);
                }
                else{
                    k++;
                }
        }
        return k;
    }
};

File: 27 Remove Element/removeElement.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        k = 0
        
        for i in range(len(nums)):
            if nums[i] != val:
                if i != k :
                    nums[i],nums[k] = nums[k],nums[i]
                    k += 1
                else:
                    k += 1
        
        return k